frontend/src/modules/SimulationTimeSeriesMatrix/view.tsx

```diff
@@ -240,8 +270,64 @@ export const View = ({ moduleContext, workbenchSession, workbenchSettings }: Mod
         subplotBuilder.addObservationsTraces(loadedVectorSpecificationsAndObservationData);
     }
 
+    function handleClickInChart(e: PlotMouseEvent) {
+        const clickedPoint: PlotDatum = e.points[0];
+        if (!clickedPoint) {
+            return;
+        }
+
+        if (clickedPoint.pointIndex >= 0 && clickedPoint.pointIndex < clickedPoint.data.x.length) {
+            const timestampUtcMs = clickedPoint.data.x[clickedPoint.pointIndex];
+            if (typeof timestampUtcMs === "number") {
+                setActiveTimestampUtcMs(timestampUtcMs);
+            }
+        }
+    }
+
     const doRenderContentError = hasRealizationsQueryError || hasStatisticsQueryError;
+
     const plotData = subplotBuilder.createPlotData();
+    const plotLayout = subplotBuilder.createPlotLayout();
+
+    const timeAnnotation: Partial<Annotations> = activeTimestampUtcMs
+        ? {
+              xref: "x",
+              yref: "paper",
+              x: activeTimestampUtcMs,
+              y: 0 - 22 / wrapperDivSize.height,
+              text: timestampUtcMsToCompactIsoString(activeTimestampUtcMs),
+              showarrow: false,
+              arrowhead: 0,
+              bgcolor: "rgba(255, 255, 255, 1)",
+              bordercolor: "rgba(255, 0, 0, 1)",
+              borderwidth: 2,
+              borderpad: 4,
+          }
+        : {};
+
+    const timeShape: Partial<Shape> = activeTimestampUtcMs
+        ? {
+              type: "line",
+              xref: "x",
+              yref: "paper",
+              x0: activeTimestampUtcMs,
+              y0: 0,
+              x1: activeTimestampUtcMs,
+              y1: 1,
+              line: {
+                  color: "red",
+                  width: 3,
+                  dash: "dot",
+              },
+          }
+        : {};
+
+    const adjustedPlotLayout: Partial<Layout> = {
+        ...plotLayout,
+        shapes: [...(plotLayout.shapes ?? []), timeShape],
+        annotations: [...(plotLayout.annotations ?? []), timeAnnotation],
+    };
```

Comment:
Only create the annotations if plotLayout.shapes !== undefined or timeShape is nonempty. As of now the default view without any selected vectors gives arrays of empty object `{}`: `shapes = [{}]` and `annotations = [{}]`. Resulting in following view:

![image](https://github.com/equinor/webviz/assets/82032112/c459ea96-cad1-4732-b0bd-0f4e4a1fa021)